src/sparseml/optim/manager.py

```diff
@@ -561,6 +561,67 @@ def qat_active(self, epoch: float) -> bool:
             else False
         )
 
+    def phase(self, epoch: float) -> Optional[str]:
+        """
+        Computes the phase that the modifiers are in.
+
+        Example usage:
+
+        ```python
+        phase = BaseManager.compose_staged(
+            manager,
+            checkpoint_manager
+        ).phase()
+        if phase is not None:
+            checkpoint_name = "best_{phase}.pt"
+        ```
+
+        :return: One of the following strings based on the pruning and quantization
+            modifiers that this Manager contains.
+
+            0. None - if either pruning or quantization is currently in progress
+            1. "dense" - if no pruning or quantization
+            2. "dense_quantized" - if only quantization
+            3. "pruned" - if only pruning
+            4. "pruned_quantized" - if both pruning and quantization
+            5. "quantized_pruned" - if quantization was before pruning
+        """
+        pruners: List[BaseModifier] = self.pruning_modifiers
+        quantizers: List[BaseModifier] = self.quantization_modifiers
+
+        if len(pruners) == 0:
+            pruned = False
+            pruning_in_progress = False
+        else:
+            pruning_start = min(mod.start_epoch for mod in pruners)
+            pruning_end = max(mod.end_epoch for mod in pruners)
+            pruning_in_progress = pruning_start <= epoch <= pruning_end
```

Comment:
Should this be:
```suggestion
            pruning_in_progress = pruning_start <= epoch < pruning_end
```
?

Reply:
has `epoch` already been applied? pruning is "complete" after the pruning modifier performs its `update` when `epoch==end_epoch`

Reply:
The intention is yes - although I guess that depends on when this is called currently.